File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/wardsupply/shared/wardsupply-asset-requisition.model.ts
import {FormGroup,FormBuilder,Validators} from '@angular/forms'
import { WardSupplyAssetRequisitionItemsModel } from './wardsupply-asset-requisitionItems.model';

export class WardSupplyAssetRequisitionModel {
    public RequisitionId: number = 0;
    public RequisitionDate: string = '';
    public CreatedBy: number = 0;
    public CreatedOn: string = '';
    public RequisitionStatus = '';
    public IssueNo =0; 
    public StoreId = 0;
    public SubStoreId = 0;
    public ModifiedOn = '';
    public ModifiedBy = 0;
    public IsCancel = 0;
    public CancelRemarks = '';
    public RequisitionNo = '';
    public Remarks = '';
    public RequisitionValidator: FormGroup = null;
    public RequisitionItemsList: Array<WardSupplyAssetRequisitionItemsModel> = new Array<WardSupplyAssetRequisitionItemsModel>();
    public VerificationId = 0;
    public EmpFullName:string='';
    constructor() {

        var _formBuilder = new FormBuilder();
        this.RequisitionValidator = _formBuilder.group({
          'StoreId': ['', Validators.compose([Validators.required])]
        });
      }
    
      public IsDirty(fieldName): boolean {
        if (fieldName == undefined)
          return this.RequisitionValidator.dirty;
        else
          return this.RequisitionValidator.controls[fieldName].dirty;
      }
    
    
      public IsValid(): boolean { if (this.RequisitionValidator.valid) { return true; } else { return false; } } public IsValidCheck(fieldName, validator): boolean {
        if (fieldName == undefined) {
          return this.RequisitionValidator.valid;
        }
        else
          return !(this.RequisitionValidator.hasError(validator, fieldName));
      }
}   

File: Code/Components/DanpheEMR.ServerModel/WardSupplyModels/WARDSupplyAssetRequisitionModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DanpheEMR.ServerModel
{
    public class WARDSupplyAssetRequisitionModel
    {
        [Key]
        public int RequisitionId { get; set; }
        public DateTime RequisitionDate { get; set; }
        public int? CreatedBy { get; set; }
        public DateTime? CreatedOn { get; set; }
        public string RequisitionStatus { get; set; }
        public int? IssueNo { get; set; }
        public int? StoreId { get; set; }
        public int? SubStoreId { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public int ModifiedBy { get; set; }
        public bool? IsCancel { get; set; }
        public string CancelRemarks { get; set; }
        public int? RequisitionNo { get; set; }
        public string Remarks { get; set; }
        public bool? IsDirectDispatch { get; set; }        

        [NotMapped]
        public virtual List<WARDSupplyAssetRequisitionItemsModel> RequisitionItemsList { get; set; }
        [NotMapped]
        public int MaxVerificationLevel { get; set; }
        [NotMapped]
        public string StoreName { get; set; }
        [NotMapped]
        public int? VerificationId { get; set; }
    }
}
